runtime/v2/manager.go

```diff
@@ -139,14 +141,118 @@ func (m *MM) ValidateGenesis(genesisData map[string]json.RawMessage) error {
 	return nil
 }
 
-// InitGenesis performs init genesis functionality for modules.
-func (m *MM) InitGenesis() {
-	panic("implement me")
+func (m *MM) InitGenesis(ctx context.Context, genesisData map[string]json.RawMessage) ([]appmodulev2.ValidatorUpdate, error) {
+	var validatorUpdates []appmodulev2.ValidatorUpdate
```

Comment:
In our call on Wednesday we talked about not returning ValidatorUpdates all the way back up the stack, I drafted something similar initially in https://github.com/cosmos/cosmos-sdk/commit/3ea653e3d093ff2a0c1d17473515873d331669c3 but will likely abandon it. 

Our conclusion is that we'd like comet/abci.go to query state for the validator updates instead of making it part of the InitGenessis API.

At the very least I think we should see how this InitGenesis call propagates up through the stack before committing this API.

Reply:
cc @julienrbrt

Reply:
TIL (wasn't on Wednesday call) :D

Reply:
Doing this may lead to discrepancies between baseapp and server/v2 btw.
Where does the validate updates get _commited_ at init genesis?